src/services/session.service.ts

```diff
@@ -21,13 +23,75 @@ export class SessionService {
 
   async signUp(user: User) {
     try {
-      const registeredUser  = this.userService.createUser(user);
+      user.isVerified = false;
+      const registeredUser  = await this.userService.createUser(user);
+
+      this.sendVerificationToken({userId: registeredUser.id, email: registeredUser.email});
+
       return registeredUser;
     } catch (error) {
       throw new DatabaseError(ErrorsMessages.USER_ALREADY_EXISTS);
     }
   }
 
+  async sendVerificationToken(tokenData: AuthInterface.ITokenDataInput) {
+      const token = await this.jwtService.createJWT(tokenData);
+      this.redisService.addTokenToVerificationList({
+        email: tokenData.email,
+        token
+      });
+      
+      EmailService.sendEmail({
+        from: APPLICATION_EMAIL as string,
+        to: tokenData.email,
+        subject: 'Verify your MVD account',
+        text: token
+      });
+  }
+
+  async verifyUser(token: string) {
+    try {
+      //decode token and get email
+      const tokenDecoded = await this.jwtService.decodeJWT(token);
+      if (tokenDecoded === null) {
+        throw new HttpError(
+          HttpStatusCode.UNAUTHORIZED,
+          'Invalid Token'
+        );
+      };
+      const email: string = (tokenDecoded as AuthInterface.ITokenPayload).data.email;
+      const id: number = (tokenDecoded as AuthInterface.ITokenPayload).data.userId;
+
+      //check if email is in list
+      const isValidEmailToVerify = await this.redisService.get(email);
+      if (!isValidEmailToVerify) {
+        this.sendVerificationToken({userId: id, email});
+        throw new HttpError(
+          HttpStatusCode.UNAUTHORIZED,
+          'A new link has been sent to your email'
+        );
+      }
+
+      //check if user with email exists on database
+      const user = await this.userService.findUserByEmail(email);
+      if (!user && !!(user as User).isVerified) {
+        throw new HttpError(
+          HttpStatusCode.UNAUTHORIZED,
+          'Invalid Token'
+        );
+      }
+      
+      //update user as verified
+      user.isVerified = true;
+      this.redisService.removeVerificationToken(user.email);
+      this.userService.editUser(user.id, user);

```

Comment:
are these ones async functions right? You should put await before of them in order to await their execution to complete and also be able to catch any error if needed